VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetails.tsx

```diff
@@ -117,11 +118,13 @@ const PrescriptionDetails: FC<PrescriptionDetailsProps> = ({ route, navigation }
               text: t('prescriptions.refill.RequestRefillButtonTitle', { count: 1 }),
               onPress: () => {
                 // Call refill request so its starts the loading screen and then go to the modal
-                if (!prescriptionInDowntime) {
-                  logAnalyticsEvent(Events.vama_rx_request_confirm(prescriptionIds))
-                  dispatch(requestRefills([prescription]))
+                if (waygateNativeAlert('WG_RefillScreenModal')) {
+                  if (!prescriptionInDowntime) {
+                    logAnalyticsEvent(Events.vama_rx_request_confirm(prescriptionIds))
+                    dispatch(requestRefills([prescription]))
+                  }
+                  navigation.navigate('RefillScreenModal')
```

Comment:
Thinking about your mention of navigation.navigate in the other PR, ideally we'd standardize on a single navigation function/hook. Not sure why we sometimes use navigation.navigate and sometimes navigateTo

Reply:
navigateTo doesn't allow for analytics/on press functions. navigation.navigate does

Reply:
Can't we just do 

```
const fileRequestsPress = () => {
  logAnalyticsEvent(Events.vama_claim_review(claimID, attributes.claimType, count))
  navigateTo('FileRequest', { claimID })()
}
```

which immediately invokes the function returned by navigateTo? Or is there something else I'm missing?

Reply:
@alexandec That seems like it should work, because `navigateTo` is just the function that's returned from the [useRouteNavigation](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/3faf9839110103962a669475a95b1962014595f2/VAMobile/src/utils/hooks/index.tsx#L91-L105) hook, which also calls `navigation.navigate` under the hood. Like you mentioned, it'd be nice if we could standardize how we navigate in the app so that we only use a single hook. There shouldn't be a race condition between logging the event and navigating to the screen, since this is happening synchronously. Unless we're missing something 🤔 @Sparowhawk?